packages/actor-query-operation-leftjoin-nestedloop/lib/ActorQueryOperationLeftJoinNestedLoop.ts

```diff
@@ -33,12 +48,14 @@ export class ActorQueryOperationLeftJoinNestedLoop extends ActorQueryOperationTy
         rightStream.on('data', (rightItem) => {
           const join = ActorRdfJoin.join(leftItem, rightItem);
           if (join) {
-            // if (pattern.expression) {
-            //   // TODO: do this once expressions are implemented
-            //   // Values of both streams can be used for this expression
-            // }
-
-            bindingsStream._push(join);
+            // If we want to keep orderening, next() should be used here.
```

Comment:
Ordering is not required, so this shouldn't be a problem.

But the `next()` call can indeed cause problems. I suggest implementing a CountDownLatch for calling `next()`.